app/grandchallenge/components/views.py

```diff
@@ -270,6 +276,29 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class CIVSetDetail(
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    DetailView,
+):
+    model = None
+    permission_required = None
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+    template_name = "components/civ_set_detail.html"
+
+    @cached_property
+    def base_object(self):
+        return NotImplementedError

```

Comment:
Since this is the same for both views, you could save yourself overriding this property on the subclassed views. What do you think? 

```suggestion
        return self.object.base_object

```